<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.550">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Monzer Sophie Zack">
<meta name="dcterms.date" content="2024-05-14">

<title>quarto5 - Final Project</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<script src="site_libs/plotly-binding-4.10.4/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="site_libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">quarto5</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./lab.html"> 
<span class="menu-text">Labs</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./projects.html"> 
<span class="menu-text">Projects</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./Final.html" aria-current="page"> 
<span class="menu-text">Final Project</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#abstract" id="toc-abstract" class="nav-link active" data-scroll-target="#abstract">Abstract</a></li>
  <li><a href="#motovating-reasons" id="toc-motovating-reasons" class="nav-link" data-scroll-target="#motovating-reasons">Motovating Reasons</a></li>
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#methods" id="toc-methods" class="nav-link" data-scroll-target="#methods">Methods</a>
  <ul class="collapse">
  <li><a href="#this-code-was-used-to-load-the-data" id="toc-this-code-was-used-to-load-the-data" class="nav-link" data-scroll-target="#this-code-was-used-to-load-the-data">This code was used to load the data:</a></li>
  <li><a href="#this-code-was-run-to-help-filter-the-data" id="toc-this-code-was-run-to-help-filter-the-data" class="nav-link" data-scroll-target="#this-code-was-run-to-help-filter-the-data">This code was run to help filter the data:</a></li>
  <li><a href="#this-code-used-to-generate-data-files-for-sankey-plots" id="toc-this-code-used-to-generate-data-files-for-sankey-plots" class="nav-link" data-scroll-target="#this-code-used-to-generate-data-files-for-sankey-plots">This Code Used to Generate Data Files for Sankey Plots</a></li>
  <li><a href="#preparing-code-for-phylogenic-trees" id="toc-preparing-code-for-phylogenic-trees" class="nav-link" data-scroll-target="#preparing-code-for-phylogenic-trees">Preparing Code for Phylogenic Trees</a></li>
  </ul></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">Results</a>
  <ul class="collapse">
  <li><a href="#yellowstone" id="toc-yellowstone" class="nav-link" data-scroll-target="#yellowstone">Yellowstone</a></li>
  <li><a href="#gemmatimonadota" id="toc-gemmatimonadota" class="nav-link" data-scroll-target="#gemmatimonadota">Gemmatimonadota</a></li>
  </ul></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion">Discussion</a>
  <ul class="collapse">
  <li><a href="#yellowstone-1" id="toc-yellowstone-1" class="nav-link" data-scroll-target="#yellowstone-1">Yellowstone</a></li>
  <li><a href="#gemmatimonadota-1" id="toc-gemmatimonadota-1" class="nav-link" data-scroll-target="#gemmatimonadota-1">Gemmatimonadota</a></li>
  </ul></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion">Conclusion</a></li>
  <li><a href="#refrences" id="toc-refrences" class="nav-link" data-scroll-target="#refrences">Refrences</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Final Project</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Monzer Sophie Zack </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">May 14, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<pre class="{r}a}"><code>if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("treeio")
BiocManager::install("ggtreeExtra")</code></pre>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Loading all the libraries</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggtree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>ggtree v3.10.1 For help: https://yulab-smu.top/treedata-book/

If you use the ggtree package suite in published research, please cite
the appropriate paper(s):

Guangchuang Yu, David Smith, Huachen Zhu, Yi Guan, Tommy Tsan-Yuk Lam.
ggtree: an R package for visualization and annotation of phylogenetic
trees with their covariates and other associated data. Methods in
Ecology and Evolution. 2017, 8(1):28-36. doi:10.1111/2041-210X.12628

Guangchuang Yu.  Data Integration, Manipulation and Visualization of
Phylogenetic Trees (1st edition). Chapman and Hall/CRC. 2022,
doi:10.1201/9781003279242

Guangchuang Yu. Using ggtree to visualize data on tree-like structures.
Current Protocols in Bioinformatics. 2020, 69:e96. doi:10.1002/cpbi.96

Attaching package: 'ggtree'

The following object is masked from 'package:tidyr':

    expand</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(TDbook) <span class="co">#A Companion Package for the Book "Data Integration, Manipulation and Visualization of Phylogenetic Trees" by Guangchuang Yu (2022, ISBN:9781032233574).</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggimage)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rphylopic)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>You are using rphylopic v.1.4.0. Please remember to credit PhyloPic contributors (hint: `get_attribution()`) and cite rphylopic in your work (hint: `citation("rphylopic")`).

Attaching package: 'rphylopic'

The following object is masked from 'package:ggimage':

    geom_phylopic</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(treeio)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>treeio v1.26.0 For help: https://yulab-smu.top/treedata-book/

If you use the ggtree package suite in published research, please cite
the appropriate paper(s):

LG Wang, TTY Lam, S Xu, Z Dai, L Zhou, T Feng, P Guo, CW Dunn, BR
Jones, T Bradley, H Zhu, Y Guan, Y Jiang, G Yu. treeio: an R package
for phylogenetic tree input and output with richly annotated and
associated data. Molecular Biology and Evolution. 2020, 37(2):599-603.
doi: 10.1093/molbev/msz240

Guangchuang Yu. Using ggtree to visualize data on tree-like structures.
Current Protocols in Bioinformatics. 2020, 69:e96. doi:10.1002/cpbi.96

Guangchuang Yu.  Data Integration, Manipulation and Visualization of
Phylogenetic Trees (1st edition). Chapman and Hall/CRC. 2022,
doi:10.1201/9781003279242</code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidytree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>If you use the ggtree package suite in published research, please cite
the appropriate paper(s):

Guangchuang Yu, David Smith, Huachen Zhu, Yi Guan, Tommy Tsan-Yuk Lam.
ggtree: an R package for visualization and annotation of phylogenetic
trees with their covariates and other associated data. Methods in
Ecology and Evolution. 2017, 8(1):28-36. doi:10.1111/2041-210X.12628

Guangchuang Yu.  Data Integration, Manipulation and Visualization of
Phylogenetic Trees (1st edition). Chapman and Hall/CRC. 2022,
doi:10.1201/9781003279242

Attaching package: 'tidytree'

The following object is masked from 'package:treeio':

    getNodeNum

The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ape)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ape'

The following objects are masked from 'package:tidytree':

    drop.tip, keep.tip

The following object is masked from 'package:treeio':

    drop.tip

The following object is masked from 'package:ggtree':

    rotate

The following object is masked from 'package:dplyr':

    where</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(TreeTools)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'TreeTools'

The following object is masked from 'package:tidytree':

    MRCA

The following object is masked from 'package:treeio':

    MRCA

The following object is masked from 'package:ggtree':

    MRCA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phytools)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: maps

Attaching package: 'maps'

The following object is masked from 'package:purrr':

    map


Attaching package: 'phytools'

The following object is masked from 'package:TreeTools':

    as.multiPhylo

The following object is masked from 'package:treeio':

    read.newick</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggnewscale)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggtreeExtra)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>ggtreeExtra v1.12.0 For help: https://yulab-smu.top/treedata-book/

If you use the ggtree package suite in published research, please cite
the appropriate paper(s):

S Xu, Z Dai, P Guo, X Fu, S Liu, L Zhou, W Tang, T Feng, M Chen, L
Zhan, T Wu, E Hu, Y Jiang, X Bo, G Yu. ggtreeExtra: Compact
visualization of richly annotated phylogenetic data. Molecular Biology
and Evolution. 2021, 38(9):4039-4042. doi: 10.1093/molbev/msab166</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggstar)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'plotly'

The following object is masked from 'package:ggplot2':

    last_plot

The following object is masked from 'package:stats':

    filter

The following object is masked from 'package:graphics':

    layout</code></pre>
</div>
</div>
<section id="abstract" class="level1">
<h1>Abstract</h1>
<p>Using what was learned in Evolutionary Genomics we analyzed bacterium diversity and evolutionary lineage at a specific site in Yellowstone National Park. We also looked and diversity, behavior and evolutionary lineage within the specific phylum Gemmatimonadota.</p>
</section>
<section id="motovating-reasons" class="level1">
<h1>Motovating Reasons</h1>
<p>This report is partly a demonstration of the skills learned throughout the course. While I am not the most proficient in R, the data generated here can be applied to real world situations. I wanted to understand the diversity of bacteria that live in Yellowstone National Park as well as the populations and make up of those bacteria. I also wanted to learn more about a particular phylum of bacteria gemmatimonadota, specifically about their evolutionary lineage and their diversity within the phylum.</p>
</section>
<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>Lab group members Monzer, Sophie and Zack were assigned Yellowstone, a NEON site location, and subsequently were assigned to Gemmatimonadota, a phylum of bacteria. As mentioned, Yellowstone National Park (YELL) is a terrestrial NEON site, that is located in Bozeman, MT. Yellowstone consists of rolling hills, and is characterized by a pine-dominated forest with sage and grass wetlands. YELL is highly representative of a wildland area in NEON’s Northern Rockies Domain (D12) and the trophic structure and community interactions are probably more representative of those that were widespread in the region before Euro-American influence than any other place in the Domain. Thus, the site offers a rare opportunity to understand interactions among climate, natural disturbance, ecosystem processes, and community structure in integrated terrestrial and aquatic systems that are representative of those of intact wildlands across the Domain. Gemmatimonadota are a phylum of bacteria established in the year 2003 after it was discovered in a sewage treatment system. They are a gram negative bacterium, can grow using both aerobic and anaerobic respiration, and studies show that they make up about 2% of soil bacterial communities. The phylum contains two classes: Gemmatimonadetes and Longimicrobia. As for where Gemmatimonadota are found, they are distributed in many natural habitats, but they seem to prefer drier soils as they are well adapted to live in low moisture environments. They make up about 2% of soil bacterial communities and have been identified as one of the top nine phyla found in soils; yet, there are currently only six cultured isolates. The phylum Gemmatimonadota contains two classes, Gemmatimonadetes and Longimicrobia. This project was done in concordance with the course Biology 476/676 Evolutionary Genomics &amp; Bioinformatics taught by Professor Jeffrey Blanchard. This course introduces life science students to evolutionary genomics, bioinformatics, and data sciences. Through computer-based lab sessions, students gain skills in Unix command line, R, reproducible research, and cloud computing. These labs also cover DNA sequence searches, sequence alignment, variation detection, phylogenetics, comparative genomics, and genome visualization. The purpose of this final project was to apply the skills acquired in the course to a real-life scenario, specifically focusing on studying the assigned phylum Gemmatimonadota at the Yellowstone National Park site.</p>
</section>
<section id="methods" class="level1">
<h1>Methods</h1>
<p>In this project, we utilized the Posit Cloud and R Markdown to create graphs and tables to summarize the analyzed data. The NEON metagenome data, collected from the designated sources, underwent thorough analysis using R Studio and R Markdown to extract meaningful insights. Furthermore, we employed phylogeny to handle data downloaded from the Joint Genome Institute (JGI), specifically referencing the GOLD Study ID Gs0161344 for this project.</p>
<section id="this-code-was-used-to-load-the-data" class="level2">
<h2 class="anchored" data-anchor-id="this-code-was-used-to-load-the-data">This code was used to load the data:</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_Ind <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"data/NEON/GOLD_Study_ID_Gs0161344_NEON.csv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span> <span class="sc">!=</span> <span class="st">"NEON combined assembly"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 1754 Columns: 19
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr   (8): Bin ID, Genome Name, Bin Quality, Bin Lineage, GTDB-Tk Taxonomy L...
dbl  (10): IMG Genome ID, Bin Completeness, Bin Contamination, Total Number ...
date  (1): Date Added

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_Ind_tax <span class="ot">&lt;-</span> NEON_MAGs_Ind <span class="sc">%&gt;%</span> </span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Domain"</span>, <span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>, <span class="st">"Family"</span>, <span class="st">"Genus"</span>), <span class="st">"; "</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 6 pieces. Additional pieces discarded in 21 rows [12, 32, 66, 79, 80,
88, 96, 102, 104, 240, 334, 386, 657, 790, 846, 931, 943, 983, 1041, 1095,
...].</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 6 pieces. Missing pieces filled with `NA` in 282 rows [6, 7, 42, 49,
50, 55, 60, 83, 85, 97, 100, 105, 107, 113, 114, 116, 119, 125, 129, 130, ...].</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"data/NEON/GOLD_Study_ID_Gs0161344_NEON_edArchaea.csv"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove columns that are not needed for data analysis</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">`</span><span class="at">GOLD Study ID</span><span class="st">`</span>, <span class="st">`</span><span class="at">Bin Methods</span><span class="st">`</span>, <span class="st">`</span><span class="at">Created By</span><span class="st">`</span>, <span class="st">`</span><span class="at">Date Added</span><span class="st">`</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># create a new column with the Assembly Type</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">"Assembly Type"</span> <span class="ot">=</span> <span class="fu">case_when</span>(<span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span> <span class="sc">==</span> <span class="st">"NEON combined assembly"</span> <span class="sc">~</span> <span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span>,</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>                            <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Individual"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Assembly Type"</span>, str_replace, <span class="st">"NEON combined assembly"</span>, <span class="st">"Combined"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Domain"</span>, <span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>, <span class="st">"Family"</span>, <span class="st">"Genus"</span>), <span class="st">"; "</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get rid of the the common string "Soil microbial communities from "</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Genome Name"</span>, str_replace, <span class="st">"Terrestrial soil microbial communities from "</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Use the first `-` to split the column in two</span></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Site"</span>,<span class="st">"Sample Name"</span>), <span class="st">" - "</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get rid of the the common string "S-comp-1"</span></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Sample Name"</span>, str_replace, <span class="st">"-comp-1"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># separate the Sample Name into Site ID and plot info</span></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">Sample Name</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Site ID"</span>,<span class="st">"subplot.layer.date"</span>), <span class="st">"_"</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>,) <span class="sc">%&gt;%</span> </span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># separate the plot info into 3 columns</span></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">subplot.layer.date</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Subplot"</span>, <span class="st">"Layer"</span>, <span class="st">"Date"</span>), <span class="st">"-"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 1754 Columns: 19
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr   (8): Bin ID, Genome Name, Bin Quality, Bin Lineage, GTDB-Tk Taxonomy L...
dbl  (10): IMG Genome ID, Bin Completeness, Bin Contamination, Total Number ...
date  (1): Date Added

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 6 pieces. Additional pieces discarded in 46 rows [3, 4, 24, 25, 26,
27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 54, 232, 267, ...].</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 6 pieces. Missing pieces filled with `NA` in 446 rows [1, 2, 9, 10,
11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 46, 50, 53, ...].</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 2 pieces. Missing pieces filled with `NA` in 624 rows [4, 7, 8, 236,
237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252,
...].</code></pre>
</div>
</div>
</section>
<section id="this-code-was-run-to-help-filter-the-data" class="level2">
<h2 class="anchored" data-anchor-id="this-code-was-run-to-help-filter-the-data">This code was run to help filter the data:</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filters for Yellowstone</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_Yellow <span class="ot">&lt;-</span> NEON_MAGs <span class="sc">%&gt;%</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at">Site ID</span><span class="st">`</span><span class="sc">==</span><span class="st">"YELL"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filters for Gemmatimonadota</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_Gemmatimonadota <span class="ot">&lt;-</span>NEON_MAGs <span class="sc">%&gt;%</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Phylum<span class="sc">==</span><span class="st">"Gemmatimonadota"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>NEON_chemistry <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/NEON/neon_plot_soilChem1_metadata.tsv"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove -COMP from genomicsSampleID</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"genomicsSampleID"</span>, str_replace, <span class="st">"-COMP"</span>, <span class="st">""</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 87 Columns: 17
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr   (5): genomicsSampleID, siteID, plotID, nlcdClass, horizon
dbl  (11): decimalLatitude, decimalLongitude, elevation, soilTemp, d15N, org...
date  (1): collectionDate

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_metagenomes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/NEON/exported_img_data_Gs0161344_NEON.tsv"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span> <span class="ot">=</span> <span class="st">`</span><span class="at">Genome Name / Sample Name</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(<span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span>, <span class="st">'re-annotation'</span>, <span class="at">negate =</span> T)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(<span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span>, <span class="st">'WREF plot'</span>, <span class="at">negate =</span> T)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 176 Columns: 46
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (18): Domain, Sequencing Status, Study Name, Genome Name / Sample Name, ...
dbl (16): taxon_oid, IMG Genome ID, Depth In Meters, Elevation In Meters, Ge...
lgl (12): Altitude In Meters, Chlorophyll Concentration, Longhurst Code, Lon...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a> NEON_MAGs_metagenomes<span class="ot">&lt;-</span> NEON_MAGs_metagenomes <span class="sc">%&gt;%</span> </span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get rid of the the common string "Soil microbial communities from "</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Genome Name"</span>, str_replace, <span class="st">"Terrestrial soil microbial communities from "</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Use the first `-` to split the column in two</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Site"</span>,<span class="st">"Sample Name"</span>), <span class="st">" - "</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get rid of the the common string "-comp-1"</span></span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Sample Name"</span>, str_replace, <span class="st">"-comp-1"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># separate the Sample Name into Site ID and plot info</span></span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">Sample Name</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Site ID"</span>,<span class="st">"subplot.layer.date"</span>), <span class="st">"_"</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>,) <span class="sc">%&gt;%</span> </span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># separate the plot info into 3 columns</span></span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">subplot.layer.date</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Subplot"</span>, <span class="st">"Layer"</span>, <span class="st">"Date"</span>), <span class="st">"-"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 2 pieces. Missing pieces filled with `NA` in 1 rows [53].</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>NEON_Full <span class="ot">&lt;-</span> NEON_MAGs<span class="sc">%&gt;%</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(NEON_chemistry, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Sample Name"</span> <span class="ot">=</span> <span class="st">"genomicsSampleID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(NEON_MAGs_metagenomes, <span class="at">by =</span> <span class="st">"Sample Name"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="this-code-used-to-generate-data-files-for-sankey-plots" class="level2">
<h2 class="anchored" data-anchor-id="this-code-used-to-generate-data-files-for-sankey-plots">This Code Used to Generate Data Files for Sankey Plots</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_Sankey <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"data/NEON/GOLD_Study_ID_Gs0161344_NEON_edArchaea.csv"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove columns that are not needed for data analysis</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">`</span><span class="at">GOLD Study ID</span><span class="st">`</span>, <span class="st">`</span><span class="at">Bin Methods</span><span class="st">`</span>, <span class="st">`</span><span class="at">Created By</span><span class="st">`</span>, <span class="st">`</span><span class="at">Date Added</span><span class="st">`</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># create a new column with the Assembly Type</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">"Assembly Type"</span> <span class="ot">=</span> <span class="fu">case_when</span>(<span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span> <span class="sc">==</span> <span class="st">"NEON combined assembly"</span> <span class="sc">~</span> <span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span>,</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>                            <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Individual"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Assembly Type"</span>, str_replace, <span class="st">"NEON combined assembly"</span>, <span class="st">"Combined"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Domain"</span>, <span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>, <span class="st">"Family"</span>, <span class="st">"Genus"</span>), <span class="st">"; "</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get rid of the the common string "Soil microbial communities from "</span></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Genome Name"</span>, str_replace, <span class="st">"Terrestrial soil microbial communities from "</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Use the first `-` to split the column in two</span></span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Site"</span>,<span class="st">"Sample Name"</span>), <span class="st">" - "</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get rid of the the common string "S-comp-1"</span></span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Sample Name"</span>, str_replace, <span class="st">"-comp-1"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span></span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># separate the Sample Name into Site ID and plot info</span></span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">Sample Name</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Site ID"</span>,<span class="st">"subplot.layer.date"</span>), <span class="st">"_"</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>,) <span class="sc">%&gt;%</span> </span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># separate the plot info into 3 columns</span></span>
<span id="cb41-18"><a href="#cb41-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">subplot.layer.date</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Subplot"</span>, <span class="st">"Layer"</span>, <span class="st">"Date"</span>), <span class="st">"-"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 1754 Columns: 19
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr   (8): Bin ID, Genome Name, Bin Quality, Bin Lineage, GTDB-Tk Taxonomy L...
dbl  (10): IMG Genome ID, Bin Completeness, Bin Contamination, Total Number ...
date  (1): Date Added

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 6 pieces. Additional pieces discarded in 46 rows [3, 4, 24, 25, 26,
27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 54, 232, 267, ...].</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 6 pieces. Missing pieces filled with `NA` in 446 rows [1, 2, 9, 10,
11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 46, 50, 53, ...].</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 2 pieces. Missing pieces filled with `NA` in 624 rows [4, 7, 8, 236,
237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252,
...].</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_ind_Sankey <span class="ot">&lt;-</span> NEON_MAGs_Sankey <span class="sc">%&gt;%</span> </span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at">Assembly Type</span><span class="st">`</span> <span class="sc">==</span> <span class="st">"Individual"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_co_Sankey <span class="ot">&lt;-</span> NEON_MAGs_Sankey <span class="sc">%&gt;%</span> </span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at">Assembly Type</span><span class="st">`</span> <span class="sc">==</span> <span class="st">"Combined"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select the GTDB Taxonomic lineage and separate into taxonomic levels</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> NEON_MAGs_co_Sankey <span class="sc">%&gt;%</span> </span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># NAs are likely Archaea</span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">replace_na</span>(<span class="fu">list</span>(<span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span> <span class="ot">=</span> <span class="st">'Archaea'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Pavian format requires p__ etc</span></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Domain"</span>, <span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>, <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"Species"</span>), <span class="st">"; "</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 7 pieces. Missing pieces filled with `NA` in 615 rows [2, 3, 4, 5, 6,
7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, ...].</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill in the NAs with the taxonomic the higher taxonomic level to the left</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>sankey_data[] <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(sankey_data, <span class="dv">1</span>, zoo<span class="sc">::</span>na.locf))</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Put the data into a format that can be read by the Sankey App</span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> sankey_data <span class="sc">%&gt;%</span> </span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="at">col =</span> <span class="st">"classification"</span>, <span class="fu">c</span>(Domain, Phylum, Class, Order, Family, Genus, Species), <span class="at">sep=</span><span class="st">'; '</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"Archaea"</span>, <span class="st">"d__Archaea"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"Bacteria"</span>, <span class="st">"d__Bacteria"</span>) <span class="sc">%&gt;%</span>  </span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|p__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|c__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|o__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|f__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|g__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|s__"</span>)  </span>
<span id="cb50-16"><a href="#cb50-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-17"><a href="#cb50-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Create format for Pavian with counts for each taxonomic level</span></span>
<span id="cb50-18"><a href="#cb50-18" aria-hidden="true" tabindex="-1"></a>sankey_data_s <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb50-19"><a href="#cb50-19" aria-hidden="true" tabindex="-1"></a>sankey_data_g <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb50-20"><a href="#cb50-20" aria-hidden="true" tabindex="-1"></a>sankey_data_f <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb50-21"><a href="#cb50-21" aria-hidden="true" tabindex="-1"></a>sankey_data_o <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb50-22"><a href="#cb50-22" aria-hidden="true" tabindex="-1"></a>sankey_data_c <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb50-23"><a href="#cb50-23" aria-hidden="true" tabindex="-1"></a>sankey_data_p <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb50-24"><a href="#cb50-24" aria-hidden="true" tabindex="-1"></a>sankey_data_d <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb50-25"><a href="#cb50-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-26"><a href="#cb50-26" aria-hidden="true" tabindex="-1"></a>sankey_data_g<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|s__.*"</span>, <span class="st">""</span>, sankey_data_g<span class="sc">$</span>classification)  </span>
<span id="cb50-27"><a href="#cb50-27" aria-hidden="true" tabindex="-1"></a>sankey_data_f<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|g__.*"</span>, <span class="st">""</span>, sankey_data_f<span class="sc">$</span>classification)  </span>
<span id="cb50-28"><a href="#cb50-28" aria-hidden="true" tabindex="-1"></a>sankey_data_o<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|f__.*"</span>, <span class="st">""</span>, sankey_data_o<span class="sc">$</span>classification)  </span>
<span id="cb50-29"><a href="#cb50-29" aria-hidden="true" tabindex="-1"></a>sankey_data_c<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|o__.*"</span>, <span class="st">""</span>, sankey_data_c<span class="sc">$</span>classification)  </span>
<span id="cb50-30"><a href="#cb50-30" aria-hidden="true" tabindex="-1"></a>sankey_data_p<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|c__.*"</span>, <span class="st">""</span>, sankey_data_p<span class="sc">$</span>classification)  </span>
<span id="cb50-31"><a href="#cb50-31" aria-hidden="true" tabindex="-1"></a>sankey_data_d<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|p__.*"</span>, <span class="st">""</span>, sankey_data_d<span class="sc">$</span>classification)  </span>
<span id="cb50-32"><a href="#cb50-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-33"><a href="#cb50-33" aria-hidden="true" tabindex="-1"></a>sankey_data_allTaxa <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(sankey_data_s, sankey_data_g, sankey_data_f, sankey_data_o, sankey_data_c, sankey_data_p, sankey_data_d) <span class="sc">%&gt;%</span> </span>
<span id="cb50-34"><a href="#cb50-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">as.factor</span>(classification)) <span class="sc">%&gt;%</span> </span>
<span id="cb50-35"><a href="#cb50-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(classification) <span class="sc">%&gt;%</span> </span>
<span id="cb50-36"><a href="#cb50-36" aria-hidden="true" tabindex="-1"></a><span class="co"># rename for Pavian format</span></span>
<span id="cb50-37"><a href="#cb50-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">#SampleID</span><span class="st">`</span> <span class="ot">=</span> <span class="st">`</span><span class="at">classification</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-38"><a href="#cb50-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">Metaphlan2_Analysis</span><span class="st">`</span> <span class="ot">=</span> <span class="st">`</span><span class="at">n</span><span class="st">`</span>) </span>
<span id="cb50-39"><a href="#cb50-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-40"><a href="#cb50-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Write file to input to Pavian Sankey</span></span>
<span id="cb50-41"><a href="#cb50-41" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(sankey_data_allTaxa, <span class="st">"data/NEON/NEON_MAG_co_pavian.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_ind_Gemmatimonadota <span class="ot">&lt;-</span> NEON_MAGs_ind_Sankey <span class="sc">%&gt;%</span> </span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Phylum <span class="sc">==</span> <span class="st">"Gemmatimonadota"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> NEON_MAGs_ind_Gemmatimonadota <span class="sc">%&gt;%</span> </span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># NAs are likely Archaea</span></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">replace_na</span>(<span class="fu">list</span>(<span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span> <span class="ot">=</span> <span class="st">'Archaea'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Pavian format requires p__ etc</span></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Domain"</span>, <span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>, <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"Species"</span>), <span class="st">"; "</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 7 pieces. Missing pieces filled with `NA` in 16 rows [1, 2, 3, 4, 5,
6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16].</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill in the NAs with the taxonomic the higher taxonomic level to the left</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>sankey_data[] <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(sankey_data, <span class="dv">1</span>, zoo<span class="sc">::</span>na.locf))</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Put the data into a format that can be read by the Sankey App</span></span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> sankey_data <span class="sc">%&gt;%</span> </span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="at">col =</span> <span class="st">"classification"</span>, <span class="fu">c</span>(Domain, Phylum, Class, Order, Family, Genus, Species), <span class="at">sep=</span><span class="st">'; '</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"Archaea"</span>, <span class="st">"d__Archaea"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"Bacteria"</span>, <span class="st">"d__Bacteria"</span>) <span class="sc">%&gt;%</span>  </span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|p__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|c__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-12"><a href="#cb54-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|o__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-13"><a href="#cb54-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|f__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-14"><a href="#cb54-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|g__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-15"><a href="#cb54-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|s__"</span>)  </span>
<span id="cb54-16"><a href="#cb54-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-17"><a href="#cb54-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Create format for Pavian with counts for each taxonomic level</span></span>
<span id="cb54-18"><a href="#cb54-18" aria-hidden="true" tabindex="-1"></a>sankey_data_s <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb54-19"><a href="#cb54-19" aria-hidden="true" tabindex="-1"></a>sankey_data_g <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb54-20"><a href="#cb54-20" aria-hidden="true" tabindex="-1"></a>sankey_data_f <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb54-21"><a href="#cb54-21" aria-hidden="true" tabindex="-1"></a>sankey_data_o <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb54-22"><a href="#cb54-22" aria-hidden="true" tabindex="-1"></a>sankey_data_c <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb54-23"><a href="#cb54-23" aria-hidden="true" tabindex="-1"></a>sankey_data_p <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb54-24"><a href="#cb54-24" aria-hidden="true" tabindex="-1"></a>sankey_data_d <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb54-25"><a href="#cb54-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-26"><a href="#cb54-26" aria-hidden="true" tabindex="-1"></a>sankey_data_g<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|s__.*"</span>, <span class="st">""</span>, sankey_data_g<span class="sc">$</span>classification)  </span>
<span id="cb54-27"><a href="#cb54-27" aria-hidden="true" tabindex="-1"></a>sankey_data_f<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|g__.*"</span>, <span class="st">""</span>, sankey_data_f<span class="sc">$</span>classification)  </span>
<span id="cb54-28"><a href="#cb54-28" aria-hidden="true" tabindex="-1"></a>sankey_data_o<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|f__.*"</span>, <span class="st">""</span>, sankey_data_o<span class="sc">$</span>classification)  </span>
<span id="cb54-29"><a href="#cb54-29" aria-hidden="true" tabindex="-1"></a>sankey_data_c<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|o__.*"</span>, <span class="st">""</span>, sankey_data_c<span class="sc">$</span>classification)  </span>
<span id="cb54-30"><a href="#cb54-30" aria-hidden="true" tabindex="-1"></a>sankey_data_p<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|c__.*"</span>, <span class="st">""</span>, sankey_data_p<span class="sc">$</span>classification)  </span>
<span id="cb54-31"><a href="#cb54-31" aria-hidden="true" tabindex="-1"></a>sankey_data_d<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|p__.*"</span>, <span class="st">""</span>, sankey_data_d<span class="sc">$</span>classification)  </span>
<span id="cb54-32"><a href="#cb54-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-33"><a href="#cb54-33" aria-hidden="true" tabindex="-1"></a>sankey_data_ind_GEM <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(sankey_data_s, sankey_data_g, sankey_data_f, sankey_data_o, sankey_data_c, sankey_data_p, sankey_data_d) <span class="sc">%&gt;%</span> </span>
<span id="cb54-34"><a href="#cb54-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">as.factor</span>(classification)) <span class="sc">%&gt;%</span> </span>
<span id="cb54-35"><a href="#cb54-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(classification) <span class="sc">%&gt;%</span> </span>
<span id="cb54-36"><a href="#cb54-36" aria-hidden="true" tabindex="-1"></a><span class="co"># rename for Pavian format</span></span>
<span id="cb54-37"><a href="#cb54-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">#SampleID</span><span class="st">`</span> <span class="ot">=</span> <span class="st">`</span><span class="at">classification</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-38"><a href="#cb54-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">Metaphlan2_Analysis</span><span class="st">`</span> <span class="ot">=</span> <span class="st">`</span><span class="at">n</span><span class="st">`</span>) </span>
<span id="cb54-39"><a href="#cb54-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-40"><a href="#cb54-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Write file to input to Pavian Sankey</span></span>
<span id="cb54-41"><a href="#cb54-41" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(sankey_data_ind_GEM, <span class="st">"NEON_MAG_ind_pavian.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_co_Gemmatimonadota <span class="ot">&lt;-</span> NEON_MAGs_co_Sankey <span class="sc">%&gt;%</span> </span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Phylum <span class="sc">==</span> <span class="st">"Gemmatimonadota"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> NEON_MAGs_co_Gemmatimonadota <span class="sc">%&gt;%</span> </span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># NAs are likely Archaea</span></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">replace_na</span>(<span class="fu">list</span>(<span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span> <span class="ot">=</span> <span class="st">'Archaea'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Pavian format requires p__ etc</span></span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Domain"</span>, <span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>, <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"Species"</span>), <span class="st">"; "</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 7 pieces. Missing pieces filled with `NA` in 6 rows [1, 2, 3, 4, 5,
6].</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill in the NAs with the taxonomic the higher taxonomic level to the left</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>sankey_data[] <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(sankey_data, <span class="dv">1</span>, zoo<span class="sc">::</span>na.locf))</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Put the data into a format that can be read by the Sankey App</span></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> sankey_data <span class="sc">%&gt;%</span> </span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="at">col =</span> <span class="st">"classification"</span>, <span class="fu">c</span>(Domain, Phylum, Class, Order, Family, Genus, Species), <span class="at">sep=</span><span class="st">'; '</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"Archaea"</span>, <span class="st">"d__Archaea"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"Bacteria"</span>, <span class="st">"d__Bacteria"</span>) <span class="sc">%&gt;%</span>  </span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|p__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|c__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|o__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|f__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|g__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|s__"</span>)  </span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Create format for Pavian with counts for each taxonomic level</span></span>
<span id="cb58-18"><a href="#cb58-18" aria-hidden="true" tabindex="-1"></a>sankey_data_s <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb58-19"><a href="#cb58-19" aria-hidden="true" tabindex="-1"></a>sankey_data_g <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb58-20"><a href="#cb58-20" aria-hidden="true" tabindex="-1"></a>sankey_data_f <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb58-21"><a href="#cb58-21" aria-hidden="true" tabindex="-1"></a>sankey_data_o <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb58-22"><a href="#cb58-22" aria-hidden="true" tabindex="-1"></a>sankey_data_c <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb58-23"><a href="#cb58-23" aria-hidden="true" tabindex="-1"></a>sankey_data_p <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb58-24"><a href="#cb58-24" aria-hidden="true" tabindex="-1"></a>sankey_data_d <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb58-25"><a href="#cb58-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-26"><a href="#cb58-26" aria-hidden="true" tabindex="-1"></a>sankey_data_g<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|s__.*"</span>, <span class="st">""</span>, sankey_data_g<span class="sc">$</span>classification)  </span>
<span id="cb58-27"><a href="#cb58-27" aria-hidden="true" tabindex="-1"></a>sankey_data_f<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|g__.*"</span>, <span class="st">""</span>, sankey_data_f<span class="sc">$</span>classification)  </span>
<span id="cb58-28"><a href="#cb58-28" aria-hidden="true" tabindex="-1"></a>sankey_data_o<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|f__.*"</span>, <span class="st">""</span>, sankey_data_o<span class="sc">$</span>classification)  </span>
<span id="cb58-29"><a href="#cb58-29" aria-hidden="true" tabindex="-1"></a>sankey_data_c<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|o__.*"</span>, <span class="st">""</span>, sankey_data_c<span class="sc">$</span>classification)  </span>
<span id="cb58-30"><a href="#cb58-30" aria-hidden="true" tabindex="-1"></a>sankey_data_p<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|c__.*"</span>, <span class="st">""</span>, sankey_data_p<span class="sc">$</span>classification)  </span>
<span id="cb58-31"><a href="#cb58-31" aria-hidden="true" tabindex="-1"></a>sankey_data_d<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|p__.*"</span>, <span class="st">""</span>, sankey_data_d<span class="sc">$</span>classification)  </span>
<span id="cb58-32"><a href="#cb58-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-33"><a href="#cb58-33" aria-hidden="true" tabindex="-1"></a>sankey_data_ind_GEM <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(sankey_data_s, sankey_data_g, sankey_data_f, sankey_data_o, sankey_data_c, sankey_data_p, sankey_data_d) <span class="sc">%&gt;%</span> </span>
<span id="cb58-34"><a href="#cb58-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">as.factor</span>(classification)) <span class="sc">%&gt;%</span> </span>
<span id="cb58-35"><a href="#cb58-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(classification) <span class="sc">%&gt;%</span> </span>
<span id="cb58-36"><a href="#cb58-36" aria-hidden="true" tabindex="-1"></a><span class="co"># rename for Pavian format</span></span>
<span id="cb58-37"><a href="#cb58-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">#SampleID</span><span class="st">`</span> <span class="ot">=</span> <span class="st">`</span><span class="at">classification</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb58-38"><a href="#cb58-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">Metaphlan2_Analysis</span><span class="st">`</span> <span class="ot">=</span> <span class="st">`</span><span class="at">n</span><span class="st">`</span>) </span>
<span id="cb58-39"><a href="#cb58-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-40"><a href="#cb58-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Write file to input to Pavian Sankey</span></span>
<span id="cb58-41"><a href="#cb58-41" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(sankey_data_ind_GEM, <span class="st">"NEON_MAG_co_gem.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_YELL <span class="ot">&lt;-</span> NEON_MAGs <span class="sc">%&gt;%</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at">Site ID</span><span class="st">`</span><span class="sc">==</span> <span class="st">"YELL"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> NEON_MAGs_YELL <span class="sc">%&gt;%</span> </span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># NAs are likely Archaea</span></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">replace_na</span>(<span class="fu">list</span>(<span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span> <span class="ot">=</span> <span class="st">'Archaea'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Pavian format requires p__ etc</span></span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Domain"</span>, <span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>, <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"Species"</span>), <span class="st">"; "</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 7 pieces. Missing pieces filled with `NA` in 50 rows [1, 2, 3, 4, 5,
6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ...].</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill in the NAs with the taxonomic the higher taxonomic level to the left</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>sankey_data[] <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(sankey_data, <span class="dv">1</span>, zoo<span class="sc">::</span>na.locf))</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Put the data into a format that can be read by the Sankey App</span></span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>sankey_data <span class="ot">&lt;-</span> sankey_data <span class="sc">%&gt;%</span> </span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="at">col =</span> <span class="st">"classification"</span>, <span class="fu">c</span>(Domain, Phylum, Class, Order, Family, Genus, Species), <span class="at">sep=</span><span class="st">'; '</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"Archaea"</span>, <span class="st">"d__Archaea"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb62-9"><a href="#cb62-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"Bacteria"</span>, <span class="st">"d__Bacteria"</span>) <span class="sc">%&gt;%</span>  </span>
<span id="cb62-10"><a href="#cb62-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|p__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb62-11"><a href="#cb62-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|c__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb62-12"><a href="#cb62-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|o__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb62-13"><a href="#cb62-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|f__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb62-14"><a href="#cb62-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|g__"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb62-15"><a href="#cb62-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"classification"</span>, str_replace, <span class="st">"; "</span>, <span class="st">"|s__"</span>)  </span>
<span id="cb62-16"><a href="#cb62-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-17"><a href="#cb62-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Create format for Pavian with counts for each taxonomic level</span></span>
<span id="cb62-18"><a href="#cb62-18" aria-hidden="true" tabindex="-1"></a>sankey_data_s <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb62-19"><a href="#cb62-19" aria-hidden="true" tabindex="-1"></a>sankey_data_g <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb62-20"><a href="#cb62-20" aria-hidden="true" tabindex="-1"></a>sankey_data_f <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb62-21"><a href="#cb62-21" aria-hidden="true" tabindex="-1"></a>sankey_data_o <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb62-22"><a href="#cb62-22" aria-hidden="true" tabindex="-1"></a>sankey_data_c <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb62-23"><a href="#cb62-23" aria-hidden="true" tabindex="-1"></a>sankey_data_p <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb62-24"><a href="#cb62-24" aria-hidden="true" tabindex="-1"></a>sankey_data_d <span class="ot">&lt;-</span> sankey_data</span>
<span id="cb62-25"><a href="#cb62-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-26"><a href="#cb62-26" aria-hidden="true" tabindex="-1"></a>sankey_data_g<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|s__.*"</span>, <span class="st">""</span>, sankey_data_g<span class="sc">$</span>classification)  </span>
<span id="cb62-27"><a href="#cb62-27" aria-hidden="true" tabindex="-1"></a>sankey_data_f<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|g__.*"</span>, <span class="st">""</span>, sankey_data_f<span class="sc">$</span>classification)  </span>
<span id="cb62-28"><a href="#cb62-28" aria-hidden="true" tabindex="-1"></a>sankey_data_o<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|f__.*"</span>, <span class="st">""</span>, sankey_data_o<span class="sc">$</span>classification)  </span>
<span id="cb62-29"><a href="#cb62-29" aria-hidden="true" tabindex="-1"></a>sankey_data_c<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|o__.*"</span>, <span class="st">""</span>, sankey_data_c<span class="sc">$</span>classification)  </span>
<span id="cb62-30"><a href="#cb62-30" aria-hidden="true" tabindex="-1"></a>sankey_data_p<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|c__.*"</span>, <span class="st">""</span>, sankey_data_p<span class="sc">$</span>classification)  </span>
<span id="cb62-31"><a href="#cb62-31" aria-hidden="true" tabindex="-1"></a>sankey_data_d<span class="sc">$</span>classification <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|p__.*"</span>, <span class="st">""</span>, sankey_data_d<span class="sc">$</span>classification)  </span>
<span id="cb62-32"><a href="#cb62-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-33"><a href="#cb62-33" aria-hidden="true" tabindex="-1"></a>sankey_data_ind_GEM <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(sankey_data_s, sankey_data_g, sankey_data_f, sankey_data_o, sankey_data_c, sankey_data_p, sankey_data_d) <span class="sc">%&gt;%</span> </span>
<span id="cb62-34"><a href="#cb62-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">as.factor</span>(classification)) <span class="sc">%&gt;%</span> </span>
<span id="cb62-35"><a href="#cb62-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(classification) <span class="sc">%&gt;%</span> </span>
<span id="cb62-36"><a href="#cb62-36" aria-hidden="true" tabindex="-1"></a><span class="co"># rename for Pavian format</span></span>
<span id="cb62-37"><a href="#cb62-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">#SampleID</span><span class="st">`</span> <span class="ot">=</span> <span class="st">`</span><span class="at">classification</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb62-38"><a href="#cb62-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">Metaphlan2_Analysis</span><span class="st">`</span> <span class="ot">=</span> <span class="st">`</span><span class="at">n</span><span class="st">`</span>) </span>
<span id="cb62-39"><a href="#cb62-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-40"><a href="#cb62-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Write file to input to Pavian Sankey</span></span>
<span id="cb62-41"><a href="#cb62-41" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(sankey_data_ind_GEM, <span class="st">"NEON_MAG_YELL.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="preparing-code-for-phylogenic-trees" class="level2">
<h2 class="anchored" data-anchor-id="preparing-code-for-phylogenic-trees">Preparing Code for Phylogenic Trees</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_Tree <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"data/NEON/GOLD_Study_ID_Gs0161344_NEON_2024_4_21.csv"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove columns that are not needed for data analysis</span></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">`</span><span class="at">GOLD Study ID</span><span class="st">`</span>, <span class="st">`</span><span class="at">Bin Methods</span><span class="st">`</span>, <span class="st">`</span><span class="at">Created By</span><span class="st">`</span>, <span class="st">`</span><span class="at">Date Added</span><span class="st">`</span>, <span class="st">`</span><span class="at">Bin Lineage</span><span class="st">`</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># create a new column with the Assembly Type</span></span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">"Assembly Type"</span> <span class="ot">=</span> <span class="fu">case_when</span>(<span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span> <span class="sc">==</span> <span class="st">"NEON combined assembly"</span> <span class="sc">~</span> <span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span>,</span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>                            <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Individual"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Assembly Type"</span>, str_replace, <span class="st">"NEON combined assembly"</span>, <span class="st">"Combined"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"GTDB-Tk Taxonomy Lineage"</span>, str_replace, <span class="st">"d__"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span>  </span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"GTDB-Tk Taxonomy Lineage"</span>, str_replace, <span class="st">"p__"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-10"><a href="#cb63-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"GTDB-Tk Taxonomy Lineage"</span>, str_replace, <span class="st">"c__"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-11"><a href="#cb63-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"GTDB-Tk Taxonomy Lineage"</span>, str_replace, <span class="st">"o__"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-12"><a href="#cb63-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"GTDB-Tk Taxonomy Lineage"</span>, str_replace, <span class="st">"f__"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-13"><a href="#cb63-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"GTDB-Tk Taxonomy Lineage"</span>, str_replace, <span class="st">"g__"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-14"><a href="#cb63-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"GTDB-Tk Taxonomy Lineage"</span>, str_replace, <span class="st">"s__"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span></span>
<span id="cb63-15"><a href="#cb63-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Domain"</span>, <span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>, <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"Species"</span>), <span class="st">";"</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-16"><a href="#cb63-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Domain"</span>, na_if,<span class="st">""</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-17"><a href="#cb63-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Phylum"</span>, na_if,<span class="st">""</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-18"><a href="#cb63-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Class"</span>, na_if,<span class="st">""</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-19"><a href="#cb63-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Order"</span>, na_if,<span class="st">""</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-20"><a href="#cb63-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Family"</span>, na_if,<span class="st">""</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-21"><a href="#cb63-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Genus"</span>, na_if,<span class="st">""</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-22"><a href="#cb63-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Species"</span>, na_if,<span class="st">""</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-23"><a href="#cb63-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb63-24"><a href="#cb63-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get rid of the the common string "Soil microbial communities from "</span></span>
<span id="cb63-25"><a href="#cb63-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Genome Name"</span>, str_replace, <span class="st">"Terrestrial soil microbial communities from "</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-26"><a href="#cb63-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Use the first `-` to split the column in two</span></span>
<span id="cb63-27"><a href="#cb63-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Site"</span>,<span class="st">"Sample Name"</span>), <span class="st">" - "</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb63-28"><a href="#cb63-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get rid of the the common string "S-comp-1"</span></span>
<span id="cb63-29"><a href="#cb63-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Sample Name"</span>, str_replace, <span class="st">"-comp-1"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span></span>
<span id="cb63-30"><a href="#cb63-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># separate the Sample Name into Site ID and plot info</span></span>
<span id="cb63-31"><a href="#cb63-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">Sample Name</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Site ID"</span>,<span class="st">"subplot.layer.date"</span>), <span class="st">"_"</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>,) <span class="sc">%&gt;%</span> </span>
<span id="cb63-32"><a href="#cb63-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># separate the plot info into 3 columns</span></span>
<span id="cb63-33"><a href="#cb63-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">subplot.layer.date</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Subplot"</span>, <span class="st">"Layer"</span>, <span class="st">"Date"</span>), <span class="st">"-"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 1754 Columns: 19
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr   (8): Bin ID, Genome Name, Bin Quality, Bin Lineage, GTDB-Tk Taxonomy L...
dbl  (10): IMG Genome ID, Bin Completeness, Bin Contamination, Total Number ...
date  (1): Date Added

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 2 pieces. Missing pieces filled with `NA` in 624 rows [1131, 1132,
1133, 1134, 1135, 1136, 1137, 1138, 1139, 1140, 1141, 1142, 1143, 1144, 1145,
1146, 1147, 1148, 1149, 1150, ...].</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>NEON_metagenomes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/NEON/exported_img_data_Gs0161344_NEON.tsv"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">`</span><span class="at">Domain</span><span class="st">`</span>, <span class="st">`</span><span class="at">Sequencing Status</span><span class="st">`</span>, <span class="st">`</span><span class="at">Sequencing Center</span><span class="st">`</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span> <span class="ot">=</span> <span class="st">`</span><span class="at">Genome Name / Sample Name</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(<span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span>, <span class="st">'re-annotation'</span>, <span class="at">negate =</span> T)) <span class="sc">%&gt;%</span> </span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(<span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span>, <span class="st">'WREF plot'</span>, <span class="at">negate =</span> T)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 176 Columns: 46
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (18): Domain, Sequencing Status, Study Name, Genome Name / Sample Name, ...
dbl (16): taxon_oid, IMG Genome ID, Depth In Meters, Elevation In Meters, Ge...
lgl (12): Altitude In Meters, Chlorophyll Concentration, Longhurst Code, Lon...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>NEON_metagenomes <span class="ot">&lt;-</span> NEON_metagenomes <span class="sc">%&gt;%</span> </span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get rid of the the common string "Soil microbial communities from "</span></span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Genome Name"</span>, str_replace, <span class="st">"Terrestrial soil microbial communities from "</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Use the first `-` to split the column in two</span></span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Site"</span>,<span class="st">"Sample Name"</span>), <span class="st">" - "</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get rid of the the common string "-comp-1"</span></span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"Sample Name"</span>, str_replace, <span class="st">"-comp-1"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span></span>
<span id="cb68-8"><a href="#cb68-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># separate the Sample Name into Site ID and plot info</span></span>
<span id="cb68-9"><a href="#cb68-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">Sample Name</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Site ID"</span>,<span class="st">"subplot.layer.date"</span>), <span class="st">"_"</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>,) <span class="sc">%&gt;%</span> </span>
<span id="cb68-10"><a href="#cb68-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># separate the plot info into 3 columns</span></span>
<span id="cb68-11"><a href="#cb68-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="st">`</span><span class="at">subplot.layer.date</span><span class="st">`</span>, <span class="fu">c</span>(<span class="st">"Subplot"</span>, <span class="st">"Layer"</span>, <span class="st">"Date"</span>), <span class="st">"-"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 2 pieces. Missing pieces filled with `NA` in 1 rows [53].</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>NEON_chemistry <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/NEON/neon_plot_soilChem1_metadata.tsv"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove -COMP from genomicsSampleID</span></span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="st">"genomicsSampleID"</span>, str_replace, <span class="st">"-COMP"</span>, <span class="st">""</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 87 Columns: 17
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr   (5): genomicsSampleID, siteID, plotID, nlcdClass, horizon
dbl  (11): decimalLatitude, decimalLongitude, elevation, soilTemp, d15N, org...
date  (1): collectionDate

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_metagenomes_chemistry <span class="ot">&lt;-</span> NEON_MAGs <span class="sc">%&gt;%</span> </span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(NEON_metagenomes, <span class="at">by =</span> <span class="st">"Sample Name"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(NEON_chemistry, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Sample Name"</span> <span class="ot">=</span> <span class="st">"genomicsSampleID"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"label"</span> <span class="ot">=</span> <span class="st">"Bin ID"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>tree_arc <span class="ot">&lt;-</span> <span class="fu">read.tree</span>(<span class="st">"data/NEON/gtdbtk.ar53.decorated.tree"</span>)</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>tree_bac <span class="ot">&lt;-</span> <span class="fu">read.tree</span>(<span class="st">"data/NEON/gtdbtk.bac120.decorated.tree"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make a vector with the internal node labels</span></span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>node_vector_bac <span class="ot">=</span> <span class="fu">c</span>(tree_bac<span class="sc">$</span>tip.label,tree_bac<span class="sc">$</span>node.label)</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Search for your Phylum or Class to get the node</span></span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span>(<span class="st">"Gemmatimonadota"</span>, node_vector_bac, <span class="at">value =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "'1.0:p__Gemmatimonadota; c__Gemmatimonadetes'"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">match</span>(<span class="fu">grep</span>(<span class="st">"Gemmatimonadota"</span>, node_vector_bac, <span class="at">value =</span> <span class="cn">TRUE</span>), node_vector_bac)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2485</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co"># First need to preorder tree before extracting. N</span></span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>tree_bac_preorder <span class="ot">&lt;-</span> <span class="fu">Preorder</span>(tree_bac)</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>tree_Gemmatimonadota <span class="ot">&lt;-</span> <span class="fu">Subtree</span>(tree_bac_preorder, <span class="dv">2485</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_metagenomes_chemistry_CLBJ <span class="ot">&lt;-</span> NEON_MAGs_metagenomes_chemistry <span class="sc">%&gt;%</span> </span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at">Site ID.x</span><span class="st">`</span> <span class="sc">==</span> <span class="st">"YELL"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Domain <span class="sc">==</span> <span class="st">"Bacteria"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make a vector of the MAGs labels</span></span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>CLBJ_MAGs_label <span class="ot">&lt;-</span> NEON_MAGs_metagenomes_chemistry_CLBJ<span class="sc">$</span>label</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use appropriate Bacteria or Archaea tree to select your site MAG labels</span></span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>tree_bac_CLBJ_MAGs <span class="ot">&lt;-</span><span class="fu">drop.tip</span>(tree_bac,tree_bac<span class="sc">$</span>tip.label[<span class="sc">-</span><span class="fu">match</span>(CLBJ_MAGs_label, tree_bac<span class="sc">$</span>tip.label)])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="results" class="level1">
<h1>Results</h1>
<section id="yellowstone" class="level2">
<h2 class="anchored" data-anchor-id="yellowstone">Yellowstone</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>project_groups <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/NEON/project_groups.tsv"</span>) </span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 18 Columns: 15
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (9): Domain, Sequencing Status, Study Name, Genome Name / Sample Name, S...
dbl (6): taxon_oid, IMG Genome ID, Latitude, Longitude, Genome Size  * assem...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 4%">
<col style="width: 22%">
<col style="width: 27%">
<col style="width: 8%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 5%">
<col style="width: 5%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: right;">taxon_oid</th>
<th style="text-align: left;">Domain</th>
<th style="text-align: left;">Sequencing Status</th>
<th style="text-align: left;">Study Name</th>
<th style="text-align: left;">Genome Name / Sample Name</th>
<th style="text-align: left;">Sequencing Center</th>
<th style="text-align: right;">IMG Genome ID</th>
<th style="text-align: left;">Phylum</th>
<th style="text-align: left;">Class</th>
<th style="text-align: left;">GOLD Study ID</th>
<th style="text-align: left;">Ecosystem Subtype</th>
<th style="text-align: right;">Latitude</th>
<th style="text-align: right;">Longitude</th>
<th style="text-align: right;">Genome Size * assembled</th>
<th style="text-align: right;">Gene Count * assembled</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">3300069261</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_051-M-20210705-comp-1 re-annotation</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300069261</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.95428</td>
<td style="text-align: right;">-110.5416</td>
<td style="text-align: right;">2764498655</td>
<td style="text-align: right;">6957571</td>
</tr>
<tr class="even">
<td style="text-align: right;">3300060744</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_016-M-20210708-comp-1</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300060744</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.96577</td>
<td style="text-align: right;">-110.5837</td>
<td style="text-align: right;">1035151817</td>
<td style="text-align: right;">1787875</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3300061626</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_048-M-20210707-comp-1</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300061626</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.95127</td>
<td style="text-align: right;">-110.5366</td>
<td style="text-align: right;">747543128</td>
<td style="text-align: right;">1371663</td>
</tr>
<tr class="even">
<td style="text-align: right;">3300069273</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_048-M-20210707-comp-1 re-annotation</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300069273</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.95127</td>
<td style="text-align: right;">-110.5366</td>
<td style="text-align: right;">1445675622</td>
<td style="text-align: right;">3657230</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3300069238</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_012-O-20210708-comp-1 re-annotation</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300069238</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.94461</td>
<td style="text-align: right;">-110.4337</td>
<td style="text-align: right;">3125604789</td>
<td style="text-align: right;">7412587</td>
</tr>
<tr class="even">
<td style="text-align: right;">3300069283</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_002-M-20210706-comp-1 re-annotation</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300069283</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.93247</td>
<td style="text-align: right;">-110.6349</td>
<td style="text-align: right;">1084617294</td>
<td style="text-align: right;">2762045</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3300061637</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_003-M-20210708-comp-1</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300061637</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.95478</td>
<td style="text-align: right;">-110.5332</td>
<td style="text-align: right;">1393141085</td>
<td style="text-align: right;">2453808</td>
</tr>
<tr class="even">
<td style="text-align: right;">3300069234</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_016-M-20210708-comp-1 re-annotation</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300069234</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.96577</td>
<td style="text-align: right;">-110.5837</td>
<td style="text-align: right;">1697741488</td>
<td style="text-align: right;">3962632</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3300060647</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_009-M-20210706-comp-1</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300060647</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.97031</td>
<td style="text-align: right;">-110.5019</td>
<td style="text-align: right;">1136012974</td>
<td style="text-align: right;">1986999</td>
</tr>
<tr class="even">
<td style="text-align: right;">3300061625</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_051-M-20210705-comp-1</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300061625</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.95428</td>
<td style="text-align: right;">-110.5416</td>
<td style="text-align: right;">1465893308</td>
<td style="text-align: right;">2677575</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3300061175</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_012-O-20210708-comp-1</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300061175</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.94461</td>
<td style="text-align: right;">-110.4337</td>
<td style="text-align: right;">1813525940</td>
<td style="text-align: right;">3135710</td>
</tr>
<tr class="even">
<td style="text-align: right;">3300060662</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_005-M-20210708-comp-1</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300060662</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.94838</td>
<td style="text-align: right;">-110.6314</td>
<td style="text-align: right;">773391452</td>
<td style="text-align: right;">1440274</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3300069268</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_046-M-20210705-comp-1 re-annotation</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300069268</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.95236</td>
<td style="text-align: right;">-110.5416</td>
<td style="text-align: right;">1874275737</td>
<td style="text-align: right;">4830121</td>
</tr>
<tr class="even">
<td style="text-align: right;">3300069213</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_003-M-20210708-comp-1 re-annotation</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300069213</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.95478</td>
<td style="text-align: right;">-110.5332</td>
<td style="text-align: right;">2438396634</td>
<td style="text-align: right;">5878161</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3300069252</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_005-M-20210708-comp-1 re-annotation</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300069252</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.94838</td>
<td style="text-align: right;">-110.6314</td>
<td style="text-align: right;">1634258714</td>
<td style="text-align: right;">4307086</td>
</tr>
<tr class="even">
<td style="text-align: right;">3300060729</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_046-M-20210705-comp-1</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300060729</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.95236</td>
<td style="text-align: right;">-110.5416</td>
<td style="text-align: right;">929469631</td>
<td style="text-align: right;">1698143</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3300069242</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_009-M-20210706-comp-1 re-annotation</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300069242</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.97031</td>
<td style="text-align: right;">-110.5019</td>
<td style="text-align: right;">1986905664</td>
<td style="text-align: right;">4778529</td>
</tr>
<tr class="even">
<td style="text-align: right;">3300060648</td>
<td style="text-align: left;">*Microbiome</td>
<td style="text-align: left;">Permanent Draft</td>
<td style="text-align: left;">Terrestrial soil microbial communities from various NEON sites located in USA and Puerto Rico</td>
<td style="text-align: left;">Terrestrial soil microbial communities from Yellowstone NP, Wyoming, USA - YELL_002-M-20210706-comp-1</td>
<td style="text-align: left;">DOE Joint Genome Institute (JGI)</td>
<td style="text-align: right;">3300060648</td>
<td style="text-align: left;">Environmental</td>
<td style="text-align: left;">Terrestrial</td>
<td style="text-align: left;">Gs0161344</td>
<td style="text-align: left;">Temperate forest</td>
<td style="text-align: right;">44.93247</td>
<td style="text-align: right;">-110.6349</td>
<td style="text-align: right;">559493798</td>
<td style="text-align: right;">1017974</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Table 1: This table shows the different phylums present at Yellowstone National Park as well as some information about them including genome size and gene count</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="fu">datatable</span>(</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>  NEON_MAGs_Ind_tax <span class="sc">%&gt;%</span> </span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="fu">c</span>(<span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span>, <span class="st">`</span><span class="at">Class</span><span class="st">`</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">str_detect</span>(<span class="st">`</span><span class="at">Genome Name</span><span class="st">`</span>, <span class="st">'YELL'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">count</span>(Class, <span class="at">sort =</span> <span class="cn">TRUE</span>)</span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item" id="htmlwidget-54b6d8c8af8091df45c0" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-54b6d8c8af8091df45c0">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13"],["Thermoleophilia","Alphaproteobacteria","UBA4738","Actinomycetia","Thermoanaerobaculia","Verrucomicrobiae","Blastocatellia","Vicinamibacteria","Gammaproteobacteria","Limnocylindria","Polyangia","Saccharimonadia","Rubrobacteria"],[13,5,5,4,4,4,3,3,2,2,2,2,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Class<\/th>\n      <th>n<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Class","targets":1},{"name":"n","targets":2}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>Table 2: This table shows the different Classes found in Yellowstone Nation Park as well as the number of times they were counted at that site represented by the value (n)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_Yellow<span class="sc">%&gt;%</span></span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y=</span>Phylum))<span class="sc">+</span></span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>()<span class="sc">+</span></span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Phylum Counts at Yellowstone National Park"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final_files/figure-html/unnamed-chunk-39-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Figure 1: This bar graph shows the different phylums present at Yellowstone NP and how many times they appeared in the collected samples</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_Yellow <span class="sc">%&gt;%</span>   </span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Total Number of Bases</span><span class="st">`</span>, <span class="at">y =</span> <span class="st">`</span><span class="at">Phylum</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Phylum Size at Yellowstone National Park"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final_files/figure-html/unnamed-chunk-40-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Figure 2: This graph shows the various counts of each of the phylums located in Yellowstone NP and the number of bases in their genome. Since every count is shown more popular phylums will appear more.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplotly</span>(</span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="at">data =</span> NEON_MAGs_Yellow, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Gene Count</span><span class="st">`</span>, <span class="at">y =</span> <span class="st">`</span><span class="at">Scaffold Count</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> Phylum, <span class="at">shape =</span> Phylum))<span class="sc">+</span></span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Phylum Scaffold Count vs Gene Count"</span>)</span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a> )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The shape palette can deal with a maximum of 6 discrete values because more
than 6 becomes difficult to discriminate
ℹ you have requested 7 values. Consider specifying shapes manually if you need
  that many have them.</code></pre>
</div>
<div class="cell-output-display">
<div class="plotly html-widget html-fill-item" id="htmlwidget-50e89334bc283b8c10a5" style="width:100%;height:464px;"></div>
<script type="application/json" data-for="htmlwidget-50e89334bc283b8c10a5">{"x":{"data":[{"x":[5724,2751,5638,2991,5824,4792,5086,3267,2518,4578],"y":[57,570,93,563,237,450,590,652,429,5],"text":["Gene Count: 5724<br />Scaffold Count:   57<br />Phylum: Acidobacteriota<br />Phylum: Acidobacteriota","Gene Count: 2751<br />Scaffold Count:  570<br />Phylum: Acidobacteriota<br />Phylum: Acidobacteriota","Gene Count: 5638<br />Scaffold Count:   93<br />Phylum: Acidobacteriota<br />Phylum: Acidobacteriota","Gene Count: 2991<br />Scaffold Count:  563<br />Phylum: Acidobacteriota<br />Phylum: Acidobacteriota","Gene Count: 5824<br />Scaffold Count:  237<br />Phylum: Acidobacteriota<br />Phylum: Acidobacteriota","Gene Count: 4792<br />Scaffold Count:  450<br />Phylum: Acidobacteriota<br />Phylum: Acidobacteriota","Gene Count: 5086<br />Scaffold Count:  590<br />Phylum: Acidobacteriota<br />Phylum: Acidobacteriota","Gene Count: 3267<br />Scaffold Count:  652<br />Phylum: Acidobacteriota<br />Phylum: Acidobacteriota","Gene Count: 2518<br />Scaffold Count:  429<br />Phylum: Acidobacteriota<br />Phylum: Acidobacteriota","Gene Count: 4578<br />Scaffold Count:    5<br />Phylum: Acidobacteriota<br />Phylum: Acidobacteriota"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(248,118,109,1)"}},"hoveron":"points","name":"Acidobacteriota","legendgroup":"Acidobacteriota","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2954,2539,2035,1364,1556,1996,2134,6857,3141,2726,4217,2041,2827,2791,2732,2937,2448,2768,3065,2652,3170,5473,4173],"y":[537,235,298,193,184,300,343,858,524,297,609,220,23,94,145,100,52,29,118,22,460,52,14],"text":["Gene Count: 2954<br />Scaffold Count:  537<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 2539<br />Scaffold Count:  235<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 2035<br />Scaffold Count:  298<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 1364<br />Scaffold Count:  193<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 1556<br />Scaffold Count:  184<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 1996<br />Scaffold Count:  300<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 2134<br />Scaffold Count:  343<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 6857<br />Scaffold Count:  858<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 3141<br />Scaffold Count:  524<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 2726<br />Scaffold Count:  297<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 4217<br />Scaffold Count:  609<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 2041<br />Scaffold Count:  220<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 2827<br />Scaffold Count:   23<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 2791<br />Scaffold Count:   94<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 2732<br />Scaffold Count:  145<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 2937<br />Scaffold Count:  100<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 2448<br />Scaffold Count:   52<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 2768<br />Scaffold Count:   29<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 3065<br />Scaffold Count:  118<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 2652<br />Scaffold Count:   22<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 3170<br />Scaffold Count:  460<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 5473<br />Scaffold Count:   52<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota","Gene Count: 4173<br />Scaffold Count:   14<br />Phylum: Actinobacteriota<br />Phylum: Actinobacteriota"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(196,154,0,1)","opacity":1,"size":5.6692913385826778,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(196,154,0,1)"}},"hoveron":"points","name":"Actinobacteriota","legendgroup":"Actinobacteriota","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2117,2074],"y":[262,385],"text":["Gene Count: 2117<br />Scaffold Count:  262<br />Phylum: Chloroflexota<br />Phylum: Chloroflexota","Gene Count: 2074<br />Scaffold Count:  385<br />Phylum: Chloroflexota<br />Phylum: Chloroflexota"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(83,180,0,1)","opacity":1,"size":5.6692913385826778,"symbol":"square","line":{"width":1.8897637795275593,"color":"rgba(83,180,0,1)"}},"hoveron":"points","name":"Chloroflexota","legendgroup":"Chloroflexota","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[6225,5175],"y":[474,859],"text":["Gene Count: 6225<br />Scaffold Count:  474<br />Phylum: Myxococcota<br />Phylum: Myxococcota","Gene Count: 5175<br />Scaffold Count:  859<br />Phylum: Myxococcota<br />Phylum: Myxococcota"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,192,148,1)","opacity":1,"size":5.6692913385826778,"symbol":"cross-thin-open","line":{"width":1.8897637795275593,"color":"rgba(0,192,148,1)"}},"hoveron":"points","name":"Myxococcota","legendgroup":"Myxococcota","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[1032,1171],"y":[2,8],"text":["Gene Count: 1032<br />Scaffold Count:    2<br />Phylum: Patescibacteria<br />Phylum: Patescibacteria","Gene Count: 1171<br />Scaffold Count:    8<br />Phylum: Patescibacteria<br />Phylum: Patescibacteria"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,182,235,1)","opacity":1,"size":5.6692913385826778,"symbol":"square-x-open","line":{"width":1.8897637795275593,"color":"rgba(0,182,235,1)"}},"hoveron":"points","name":"Patescibacteria","legendgroup":"Patescibacteria","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[7362,1740,8843,3636,3208,5630,3661],"y":[1216,264,1540,15,374,767,472],"text":["Gene Count: 7362<br />Scaffold Count: 1216<br />Phylum: Proteobacteria<br />Phylum: Proteobacteria","Gene Count: 1740<br />Scaffold Count:  264<br />Phylum: Proteobacteria<br />Phylum: Proteobacteria","Gene Count: 8843<br />Scaffold Count: 1540<br />Phylum: Proteobacteria<br />Phylum: Proteobacteria","Gene Count: 3636<br />Scaffold Count:   15<br />Phylum: Proteobacteria<br />Phylum: Proteobacteria","Gene Count: 3208<br />Scaffold Count:  374<br />Phylum: Proteobacteria<br />Phylum: Proteobacteria","Gene Count: 5630<br />Scaffold Count:  767<br />Phylum: Proteobacteria<br />Phylum: Proteobacteria","Gene Count: 3661<br />Scaffold Count:  472<br />Phylum: Proteobacteria<br />Phylum: Proteobacteria"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(165,138,255,1)","opacity":1,"size":5.6692913385826778,"symbol":"asterisk-open","line":{"width":1.8897637795275593,"color":"rgba(165,138,255,1)"}},"hoveron":"points","name":"Proteobacteria","legendgroup":"Proteobacteria","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[1739,3871,3697,3832],"y":[153,32,5,17],"text":["Gene Count: 1739<br />Scaffold Count:  153<br />Phylum: Verrucomicrobiota<br />Phylum: Verrucomicrobiota","Gene Count: 3871<br />Scaffold Count:   32<br />Phylum: Verrucomicrobiota<br />Phylum: Verrucomicrobiota","Gene Count: 3697<br />Scaffold Count:    5<br />Phylum: Verrucomicrobiota<br />Phylum: Verrucomicrobiota","Gene Count: 3832<br />Scaffold Count:   17<br />Phylum: Verrucomicrobiota<br />Phylum: Verrucomicrobiota"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(251,97,215,1)","opacity":1,"size":5.6692913385826778,"symbol":[null,null,null,null],"line":{"width":1.8897637795275593,"color":"rgba(251,97,215,1)"}},"hoveron":"points","name":"Verrucomicrobiota","legendgroup":"Verrucomicrobiota","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":43.762557077625573,"r":7.3059360730593621,"b":40.182648401826491,"l":48.949771689497723},"plot_bgcolor":"rgba(235,235,235,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"title":{"text":"Phylum Scaffold Count vs Gene Count","font":{"color":"rgba(0,0,0,1)","family":"","size":17.534246575342465},"x":0,"xref":"paper"},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[641.45000000000005,9233.5499999999993],"tickmode":"array","ticktext":["2500","5000","7500"],"tickvals":[2500,5000,7500],"categoryorder":"array","categoryarray":["2500","5000","7500"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"y","title":{"text":"Gene Count","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-74.900000000000006,1616.9000000000001],"tickmode":"array","ticktext":["0","500","1000","1500"],"tickvals":[0,500,1000.0000000000001,1500],"categoryorder":"array","categoryarray":["0","500","1000","1500"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"x","title":{"text":"Scaffold Count","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"title":{"text":"Phylum","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"39a24304900c":{"x":{},"y":{},"colour":{},"shape":{},"type":"scatter"}},"cur_data":"39a24304900c","visdat":{"39a24304900c":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>Figure 3: This interactive graph shows the relationship between the size of the scaffolds and the genes for each of the different phylum present at the site. The lower the scaffold count the higher the quality of the sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_Yellow <span class="sc">%&gt;%</span> </span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="st">`</span><span class="at">Gene Count</span><span class="st">`</span>, <span class="at">y=</span> <span class="st">`</span><span class="at">Bin Completeness</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> Phylum))<span class="sc">+</span></span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Phylum Completeness vs Gene Count"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final_files/figure-html/unnamed-chunk-42-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Figure 4: This graph shows which phylum have missing single-copy gene markers (Bin Completeness) compared to the gene count. The higher the bin completeness the less missing single-copy gene markers.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_Yellow <span class="sc">%&gt;%</span> </span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(Phylum) <span class="sc">|</span> <span class="fu">is.na</span>(Class) <span class="sc">|</span> <span class="fu">is.na</span>(Order) <span class="sc">|</span> <span class="fu">is.na</span>(Family) <span class="sc">|</span> <span class="fu">is.na</span>(Genus)) <span class="sc">%&gt;%</span></span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Bin Quality</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="fu">position_dodge2</span>(<span class="at">width =</span> <span class="fl">0.9</span>, <span class="at">preserve =</span> <span class="st">"single"</span>)) <span class="sc">+</span></span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()<span class="sc">+</span></span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Phylum Quality at Yellowstone National Park"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final_files/figure-html/unnamed-chunk-43-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Figure 5: This graph shows the number of High Quality (HQ) and Medium Quality (MQ) samples taken at Yellowstone NP</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree_bac_CLBJ_MAGs, <span class="at">layout=</span><span class="st">"circular"</span>)  <span class="sc">%&lt;+%</span></span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>  NEON_MAGs_metagenomes_chemistry <span class="sc">+</span></span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">color=</span>Phylum))<span class="sc">+</span></span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Phylogenetic Tree for Yellowstone National Park"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Figure 6: This figure shows the ancestry of each of the phylum present at Yellowstone National Park and how they evolved apart from each other.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_url</span>(<span class="st">"data/sankey-NEON_MAG_YELL.txt.html"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<iframe src="data/sankey-NEON_MAG_YELL.txt.html" width="672" height="400px" data-external="1">
</iframe>
</div>
<p>Figure 7: Sankey Tree For the phylum present at Yellowstone National Park</p>
</section>
<section id="gemmatimonadota" class="level2">
<h2 class="anchored" data-anchor-id="gemmatimonadota">Gemmatimonadota</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">datatable</span>(</span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a>  NEON_MAGs_Gemmatimonadota <span class="sc">%&gt;%</span></span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="fu">c</span>(<span class="st">`</span><span class="at">Site</span><span class="st">`</span>,<span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span>, <span class="st">`</span><span class="at">Gene Count</span><span class="st">`</span>, <span class="st">`</span><span class="at">Total Number of Bases</span><span class="st">`</span>, <span class="st">`</span><span class="at">Bin Quality</span><span class="st">`</span>))</span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item" id="htmlwidget-1f38a98db43978ed5541" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-1f38a98db43978ed5541">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22"],["Konza Prairie Bio Station, Kansas, USA","Santa Rita Experimental Range, Tucson, Arizona, USA","Great Basin, Onaqui, Utah, USA","Great Basin, Onaqui, Utah, USA","Great Basin, Onaqui, Utah, USA","Santa Rita Experimental Range, Tucson, Arizona, USA","Great Basin, Onaqui, Utah, USA","National Grasslands LBJ, Texas, USA","Chase Lake Wetlands, North Dakota, USA","Chase Lake Wetlands, North Dakota, USA","NEON combined assembly","NEON combined assembly","NEON combined assembly","NEON combined assembly","NEON combined assembly","Lower Teakettle, Sierra National Forest, California, USA","Lower Teakettle, Sierra National Forest, California, USA","Toolik Field Station, Alaska, USA","Santa Rita Experimental Range, Tucson, Arizona, USA","National Grasslands LBJ, Texas, USA","Chase Lake Wetlands, North Dakota, USA","NEON combined assembly"],["Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; Gemmatimonadaceae","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; GWC2-71-9; JACDDX01","Bacteria; Gemmatimonadota; Gemmatimonadetes; JACCXV01","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; GWC2-71-9; WVYX01","Bacteria; Gemmatimonadota; Gemmatimonadetes; Longimicrobiales; RSA9","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; GWC2-71-9; JACDDX01","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; GWC2-71-9; 20CM-2-65-7","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; GWC2-71-9; JACDDX01","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; GWC2-71-9; JABFSM01","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; GWC2-71-9; JAGGAG01","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; GWC2-71-9; JACDDX01","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; GWC2-71-9; JACDDX01","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; GWC2-71-9; 20CM-2-65-7","Bacteria; Gemmatimonadota; Gemmatimonadetes; JACCXV01","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; GWC2-71-9; WVYX01","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; Gemmatimonadaceae; UBA4720","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; Gemmatimonadaceae; Fen-1231","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; Gemmatimonadaceae; Fen-1231","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; Gemmatimonadaceae","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; Gemmatimonadaceae; AG2","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; GWC2-71-9; SPDF01","Bacteria; Gemmatimonadota; Gemmatimonadetes; Gemmatimonadales; GWC2-71-9; SPDF01"],[4076,2063,2976,4143,2804,1929,3206,3646,4416,2770,1785,2097,3133,2897,3898,3896,2563,3162,2461,5397,2751,2211],[3977506,1922184,3030511,4428439,2784338,1870917,3411637,3583027,4400110,2591126,1750508,2032470,3253608,2908782,4155696,4086406,2524095,3195529,2289754,5785981,2850670,2252973],["MQ","MQ","HQ","HQ","MQ","MQ","HQ","MQ","MQ","MQ","MQ","MQ","MQ","HQ","MQ","HQ","MQ","MQ","MQ","MQ","HQ","MQ"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Site<\/th>\n      <th>GTDB-Tk Taxonomy Lineage<\/th>\n      <th>Gene Count<\/th>\n      <th>Total Number of Bases<\/th>\n      <th>Bin Quality<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Site","targets":1},{"name":"GTDB-Tk Taxonomy Lineage","targets":2},{"name":"Gene Count","targets":3},{"name":"Total Number of Bases","targets":4},{"name":"Bin Quality","targets":5}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>Table 3: This table shows the sites where the bacterium Gemmatimonadota is present as well as how many samples were recovered from each site. Other information like gene and base count is also shown.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_Gemmatimonadota <span class="sc">%&gt;%</span>   </span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Site, <span class="at">y =</span> <span class="st">`</span><span class="at">GTDB-Tk Taxonomy Lineage</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()<span class="sc">+</span></span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">60</span>, <span class="at">vjust=</span><span class="dv">1</span>, <span class="at">hjust=</span><span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Gemmatimonadota Lineages at Each Site"</span>, <span class="at">y =</span> <span class="st">"Lineages"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final_files/figure-html/unnamed-chunk-47-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<p>Figure 8: This graph shows where each of the different lineages of gemmatimonadota are present at each site</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_Gemmatimonadota<span class="sc">%&gt;%</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y=</span>Site))<span class="sc">+</span></span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>()<span class="sc">+</span></span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span> <span class="st">"Gemmatimonadota Counts at Each Site"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final_files/figure-html/unnamed-chunk-48-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Figure 9: This graph shows the total counts of gemmatimonadota at each site</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a>NEON_MAGs_Gemmatimonadota <span class="sc">%&gt;%</span> </span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(Phylum) <span class="sc">|</span> <span class="fu">is.na</span>(Class) <span class="sc">|</span> <span class="fu">is.na</span>(Order) <span class="sc">|</span> <span class="fu">is.na</span>(Family) <span class="sc">|</span> <span class="fu">is.na</span>(Genus)) <span class="sc">%&gt;%</span></span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Bin Quality</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="fu">position_dodge2</span>(<span class="at">width =</span> <span class="fl">0.9</span>, <span class="at">preserve =</span> <span class="st">"single"</span>)) <span class="sc">+</span></span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()<span class="sc">+</span></span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Gemmatimonadota Quality"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final_files/figure-html/unnamed-chunk-49-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Figure 10: This graph shows the number of High Quality (HQ) and Medium Quality (MQ) samples of gemmatimonadota taken at various sites.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>NEON_Gemmatimonadota <span class="ot">&lt;-</span> NEON_Full<span class="sc">%&gt;%</span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at">Phylum</span><span class="st">`</span> <span class="sc">==</span> <span class="st">"Gemmatimonadota"</span>)</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a>NEON_Gemmatimonadota <span class="sc">%&gt;%</span>   </span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Total Number of Bases</span><span class="st">`</span>, <span class="at">y =</span> <span class="st">`</span><span class="at">Site.x</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Gemmatimonadota Base Size at Each Site"</span>, <span class="at">y =</span> <span class="st">"Site"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final_files/figure-html/unnamed-chunk-50-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Figure 11: This graph shows the size of gemmatimonadota at each site</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_url</span>(<span class="st">"data/sankey-NEON_MAG_ind_pavian.txt.html"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<iframe src="data/sankey-NEON_MAG_ind_pavian.txt.html" width="672" height="400px" data-external="1">
</iframe>
</div>
<p>Figure 12: Sankey Tree of the individual assemblies of Gemmatimonadota</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_url</span>(<span class="st">"data/sankey-NEON_MAG_co_gem.txt.html"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<iframe src="data/sankey-NEON_MAG_co_gem.txt.html" width="672" height="400px" data-external="1">
</iframe>
</div>
<p>Figure 13: Sankey Tree of the combined assemblies of Gemmatimonadota</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree_Gemmatimonadota)  <span class="sc">%&lt;+%</span></span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>  NEON_MAGs_metagenomes_chemistry <span class="sc">+</span> </span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tiplab</span>(<span class="at">size=</span><span class="dv">2</span>, <span class="at">hjust=</span><span class="sc">-</span>.<span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">color=</span><span class="st">`</span><span class="at">Ecosystem Subtype</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Phylogenetic Tree of Gemmatimonadota"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final_files/figure-html/unnamed-chunk-53-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Figure 14: This figure shows the evolutionary tree of gemmatimonadota as well as which ecosystem each lineage was found in</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>NEON_Gemmatimonadota <span class="sc">%&gt;%</span>   </span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">soilTemp</span><span class="st">`</span>, <span class="at">y =</span> <span class="st">`</span><span class="at">Site.x</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Soil Temperatures"</span>, <span class="at">x =</span> <span class="st">"Soil Temperature"</span>, <span class="at">y =</span> <span class="st">"Site"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 6 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final_files/figure-html/unnamed-chunk-54-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Figure 15: This graph shows the average soil temperatures in Celsius for each site where gemmatimonadota was found in.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> NEON_Gemmatimonadota, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Ecosystem Subtype</span><span class="st">`</span>, <span class="at">y =</span> <span class="st">`</span><span class="at">soilTemp</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color=</span>Family))<span class="sc">+</span></span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Soil Temperatures Expanded"</span>, <span class="at">y =</span> <span class="st">"Soil Temperature"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 6 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Final_files/figure-html/unnamed-chunk-55-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Figure 16: This graph shows the soil temperature for each sample as well as which ecosystem it was taken from. The family of gemmatimonadota for each sample is also shown.</p>
</section>
</section>
<section id="discussion" class="level1">
<h1>Discussion</h1>
<section id="yellowstone-1" class="level2">
<h2 class="anchored" data-anchor-id="yellowstone-1">Yellowstone</h2>
<p>The results show that a particular phylum of bacteria seems to be present the most at Yellowstone NP (Figure 1). Actinobacteriota is found in most samples significantly more than any other phylum. Actinobacteriota also has a smaller base count compared to the others (Figure 2). Looking at the phylogenetic tree we can see that Actinobacteriota is very diverse in its own phylum (Figure 6). Its also interesting to see that Chloroflexota hasn’t diverged as much from the last common ancestor shared between all the phylum. We wanted to analyze all of the phylum present in more detail and found that many of the samples were of medium quality (Figure 5). If you take a look at the scaffold count many are quiet high, which means they are of lower quality (Figure 3). To see if the larger the size of the bacterium the lower the quality we compared the two. We can see from the results that for some of the phylum the more genes present the higher the scaffold count. Proteobacteria for example follows an almost linear increase. Patescibacteria has the lowest scaffold count of any phylum, but it also has the fewest genes. Actinobacteriota has more samples with a lower scaffold count than the rest of the phylums. In general the more genes a bacterium has the higher the scaffold count and the lower the quality. We wanted to examine this further so we compared the completeness of each of the phylum to the gene count (Figure 4). This will show us if the reason why the quality is low is because as the gene count becomes larger more parts of the expected genome are missing. The results however don’t show that. The samples with the smallest gene count are also the most incomplete. More analysis needs to be done to figure out why that is the case.</p>
</section>
<section id="gemmatimonadota-1" class="level2">
<h2 class="anchored" data-anchor-id="gemmatimonadota-1">Gemmatimonadota</h2>
<p>Taking a closer look at Gemmatimonadota we can see that it lives in a wide diversity of environments. The sites where the samples were found are spread all across the United States (Figure 8 and 9). This shows the bacterium isn’t clustered in one specific region and can thrive nearly anywhere. There are more samples found in Utah compared to the other sites so maybe this could be where it is most successful. The quality of the samples taken across all the sites is almost evenly split with just a bit more being medium quality (Figure 10). Looking at Figure 11 you’ll see a wide diversity in the size of genome across the sites. The Gemmatimonadota at Tucson Arizona have fewer base pairs than than the Gemmatimonadota at LBJ in Texas. Another interesting factor is that Gemmatimonadota doesn’t have a preferred climate or temperature. The sites where it was found could be arid, dry, cold, hot, or wet (Figure 14 and 15). Looking at the phylogenetic tree in Figure 14 we can see that Gemmatimonadota was found most often in shrublands. Something interesting we found was that two of the samples 33000616338_25 and 3300060695_66 are closely related however one lives in the tundra while the other lives in a temperate forest. Looking at the soild temperatures for each of the sites in figure 16 you’ll see that there is a massive difference between the two temperatures. We believe that because these two samples are closely related and they live in very different climates that Gemmatimonadota has evolved to be temperature resistant or highly adaptable. Further studies should be conducted such as placing one sample of Gemmatimonadota that lives in warm soil and placing it in colder soil then see if it survives or dies.</p>
</section>
</section>
<section id="conclusion" class="level1">
<h1>Conclusion</h1>
<p>Seven unique phylum live in Yellowstone National Park with the most successful one appearing to be Actinobacteriota. This bacterium vary in genome size with the more successful Actinobacteriota having a smaller genome. We noticed a relationship with the larger the genome count the higher the scaffold count, which means the quality of the samples was lower. We investigated further to see if this was due to larger genomes being incomplete, however that ended up being not the case.</p>
<p>Gemmatimonadota is found across various sites in the United States, and can live in a wide variety of ecosystem subtypes. Two closely related strains can even live in two very different climates which seems to suggest that temperature doesn’t effect Gemmatimonadota as much as other bacterium. The diversity within Gemmatimonadota is also on par with other phylum, like the ones found in Yellowstone National Park.</p>
</section>
<section id="refrences" class="level1">
<h1>Refrences</h1>
<p>Yellowstone National Park NEON | NSF NEON | Open Data to Understand our Ecosystems. (n.d.). Retrieved April 7, 2024, from <a href="https://www.neonscience.org/field-sites/yell" class="uri">https://www.neonscience.org/field-sites/yell</a></p>
<p>Mujakić, I., Piwosz, K., &amp; Koblížek, M. (2022). Phylum Gemmatimonadota and Its Role in the Environment. Microorganisms, 10(1), 151. <a href="https://doi.org/10.3390/microorganisms10010151" class="uri">https://doi.org/10.3390/microorganisms10010151</a></p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>